tests/test_reply_function_async.js

```diff
@@ -56,10 +53,11 @@ test('reply should throw on error on the callback', async t => {
       callback(new Error('Database failed'))
     )
 
-  await t.rejects(got('http://example.test'), {
-    name: 'RequestError',
-    message: 'Database failed',
-  })
+  await assertRejects(
```

Comment:
We could use chai-as-promised, though since we're only using `t.rejects` in five places and there are a _lot_ of `assertRejects`, I think it makes more sense to get everything using `assertRejects` in this pass and look at chai-as-promised in a later one.